### PR TITLE
fix(statistics-service): migrate data containing dots in keys

### LINKED ISSUE #6250: Core: Remove `.` in statistic-service mongo collection
## Action Item

In #6033, we provide a strategy to replace `.` chars from the document field name of collections. The same strategy should be used for the collection used by the statistic service. 

## DoD

- [ ] [mongoexport](https://docs.mongodb.com/database-tools/mongoexport/#mongodb-binary-bin.mongoexport) and [mongoimport](https://docs.mongodb.com/database-tools/mongoimport/#mongodb-binary-bin.mongoimport) work with the `keptn-stats` collection
- [ ] Unit tests

### PR DESCRIPTION
**Pull request by warber**
closes #6250 

Introduced a migration component that takes over the migration of existing data.
Also adjusted the Dockerfile to be able to use in-memory mongodb in the tests on GH

### CODE REVIEW

File: statistics-service/db/statistics_mongodb_repo.go [hunk 1]
```diff
@@ -57,28 +63,88 @@ func (s *StatisticsMongoDBRepo) GetStatistics(from, to time.Time) ([]operations.
...
+
+	return decodedResult, nil
 }
 
 // StoreStatistics godoc
 func (s *StatisticsMongoDBRepo) StoreStatistics(statistics operations.Statistics) error {
-	err := s.getCollection()
+	encodedStats, err := encodeKeys(&statistics)
+	if err != nil {
+		return nil
```
Comment (bacherfl) [inline statistics-service/db/statistics_mongodb_repo.go:78]: is this intentional?

Comment (warber) [reply statistics-service/db/statistics_mongodb_repo.go]: nope, fixed, thx

File: statistics-service/db/statistics_mongodb_repo.go [hunk 2]
```diff
@@ -57,28 +63,88 @@ func (s *StatisticsMongoDBRepo) GetStatistics(from, to time.Time) ([]operations.
...
 
-// DeleteStatistics godoc
+// MigrateKeys goes through each and every document and escapes keys which are known to dots (".")
+// https://github.com/keptn/keptn/issues/6250
+func (s *StatisticsMongoDBRepo) MigrateKeys() (uint, error) {
+	err := s.getCollection()
+	if err != nil {
+		return 0, err
+	}
+	ctx, cancel := context.WithTimeout(context.TODO(), 10*time.Second)
```
Comment (bacherfl) [inline statistics-service/db/statistics_mongodb_repo.go:104]: will that timeout be enough for collections with a large number of entries?

**Comment by thisthat** [comment]
There is a panic error in the test `TestStatisticsMongoDBRepo_Store_And_Get_Statistics`: 
https://github.com/keptn/keptn/runs/4394678838?check_suite_focus=true#step:5:140

File: statistics-service/main.go [hunk 3]
```diff
@@ -34,11 +36,10 @@ import (
 // @BasePath /v1
 
 const envVarLogLevel = "LOG_LEVEL"
+const envDisableMigration = "DISABLE_DATA_MIGRATION"
```
Comment (thisthat) [inline statistics-service/main.go:39]: could you please document somewhere this variable?
I would also add this option to our helm values in the installation charts.

**Review by bacherfl — APPROVED** [review]

**Pull request merged**